app/vue3/src/client/preview/index.ts

```diff
@@ -24,7 +24,9 @@ function normalizeFunctionalComponent(options: ConcreteComponent): ComponentOpti
   return typeof options === 'function' ? { render: options, name: options.name } : options;
 }
 
-function prepare(story: StoryFnVueReturnType, innerStory?: ConcreteComponent): Component | null {
+function prepare(rawStory: StoryFnVueReturnType, innerStory?: ConcreteComponent): Component | null {
+  const story = rawStory as ComponentOptions;
```

Comment:
I adapted this part from https://github.com/storybookjs/storybook/blob/next/app/vue/src/client/preview/index.ts#L29 not sure if this the proper type though…